src/dos/dos_execute.cpp

```diff
@@ -556,3 +562,67 @@ bool DOS_Execute(char * name,PhysPt block_pt,uint8_t flags) {
 	}
 	return false;
 }
+
+std::optional<uint16_t> DOS_CreateFakeTsrArea(const uint32_t bytes,
+                                              const bool force_low_memory)
+{
+	constexpr uint16_t StackNeeded = 0x80;
+	constexpr uint16_t PspSegments = 0x10;
+	constexpr uint32_t MaxTsrSize  = 512 * 1024;
+
+	constexpr uint16_t CommandTailSegment = 0x08;
+	constexpr uint16_t CommandTailSize    = 0x80;
+
+	// Try to matche the smallest block
+	const uint8_t MemAllocStrategy = force_low_memory ? 0x01 : 0x81;
+
+	if (bytes == 0 || bytes > MaxTsrSize || reg_sp <= StackNeeded) {
+		return {};
+	}
+
+	// Get current DOS PSP
+	const auto app_psp_segment = dos.psp();
+	DOS_PSP psp(app_psp_segment);
+
+	// Reserve stack space for the fake process
+	reg_sp -= StackNeeded;
+
+	// Set empty DOS parameter block
+	DOS_ParamBlock param_block(SegPhys(ss) + reg_sp);
+	param_block.Clear();
+
+	// Calculate number of memory blocks to allocate
+	uint16_t blocks = PspSegments + (bytes + 15) / 16;
+
+	// Allocate memory
+	uint16_t tsr_psp_segment = 0;
+	const auto old_strategy = DOS_GetMemAllocStrategy();
+	DOS_SetMemAllocStrategy(MemAllocStrategy);
+	const auto result = DOS_AllocateMemory(&tsr_psp_segment, &blocks);
+	DOS_SetMemAllocStrategy(old_strategy);
+
+	if (!result) {
+		// Memory allocation failed
+		reg_sp += StackNeeded;
+		return {};
+	}
+
+	// Setup the PSP
+	setup_psp(tsr_psp_segment, blocks, 0);
+
+	// Copy the command tail
+	MEM_BlockCopy(PhysicalMake(tsr_psp_segment + CommandTailSegment, 0),
+	              PhysicalMake(app_psp_segment + CommandTailSegment, 0),
+	              CommandTailSize);
+
+	// Clear the TSR memory
+	const auto start_segment = tsr_psp_segment + PspSegments;
+	for (auto idx = start_segment; idx < blocks - PspSegments; idx++) {
+		mem_writeq(PhysicalMake(idx, 0), 0);
+		mem_writeq(PhysicalMake(idx, 8), 0);
```

Comment:
Can we explain what's going on here? What are the 0 and 8 magic constants accomplishing?